Rubberduck.Core/UI/CodeExplorer/Commands/ImportCommand.cs

```diff
@@ -86,9 +93,66 @@ protected override void OnExecute(object parameter)
 
             foreach (var filename in _openFileDialog.FileNames)
             {
-                using (var components = project.VBComponents)
+                var extension = Path.GetExtension(filename);
+
+                var sourceText = string.Join(Environment.NewLine, File.ReadAllLines(filename));
+                var tempHelper = (CodeExplorerItemViewModel)parameter;
+                var newFolderName = (parameter is CodeExplorerCustomFolderViewModel) ? tempHelper.Name : tempHelper.GetSelectedDeclaration().CustomFolder;
+
+                var startRule = VBACodeStringParser.Parse(sourceText, t => t.startRule());
+                
+                if (HasModuleDeclarations((Antlr4.Runtime.ParserRuleContext)startRule.parseTree, out var moduleDeclarations))
+                {
+                    if (HasOptionExplicit((Antlr4.Runtime.ParserRuleContext)startRule.parseTree, out var optionExplicitStmt))
+                    {
+                        if (HasFolderAnnotation((Antlr4.Runtime.ParserRuleContext)startRule.parseTree, out var FolderAnnotation))
+                        {
+                            var oldFolder = FolderAnnotation.GetChild<VBAParser.AnnotationArgContext>(); //update with AnnotationArgList
+                            startRule.rewriter.Replace(oldFolder.SourceInterval.a, oldFolder.SourceInterval.b, newFolderName);
```

Comment:
The rewriter has an overload taking a `ParserRuleContext` that will basically do what you have implemented here.

Reply:
I forgot that you merely have an Antlr `TokenstremRewriter` and not an `IModuleRewriter`.